src/math/numbers.c

```diff
@@ -177,3 +173,18 @@ uint32_t crc32(uint32_t base, const void *data, uint32_t bytes)
 
 	return ~crc;
 }
+
+/**
+ * \brief Calculate log2(alignment) alignment should be a power of 2.
+ * \param[in] alignment The component copy frame alignment limit.
+ */
+uint16_t log2_uint16(uint16_t alignment)
```

Comment:
Can we use the compiler builtin (find last set) or use an API linux Linux. See
https://www.kernel.org/doc/htmldocs/kernel-api/API-fls.html

Reply:
in fact, I think this would be a lot simpler overall (and in any other components) to use a table of frame size and alignment. i.e. for any given frame size we lookup the desired alignment